src/library/scala/collection/concurrent/TrieMap.scala

```diff
@@ -579,17 +579,17 @@ private[collection] final class CNode[K, V](val bitmap: Int, val array: Array[Ba
 
   /* quiescently consistent - don't call concurrently to anything involving a GCAS!! */
   private def collectElems: Seq[(K, V)] = array.flatMap({
-    case sn: SNode[K, V] => Some(sn.kvPair): IterableOnce[(K, V)]
+    case sn: SNode[K, V] => Iterable.single(sn.kvPair)
     case in: INode[K, V] => in.mainnode match {
-      case tn: TNode[K, V] => Some(tn.kvPair): IterableOnce[(K, V)]
+      case tn: TNode[K, V] => Iterable.single(tn.kvPair)
```

Comment:
`tn.kvPair :: Nil`?

Reply:
Without benchmarking I would assume that `Iterable.single` is faster because it uses a special single-element `Iterable`. Since you only need an `IterableOnce` in this case, `Iterator.single` should be even better.